### DISCUSSION TITLE
Does oxigraph (oxrdflib) currently support rdf-star syntax with Python?

### DISCUSSION THREAD

**Discussion by joylix** [oxigraph/oxigraph#886] [Q&A]
I would like to know if oxrdflib currently supports reading and writing of rdf-star(quoted triple) turtle files with Python?   If so, how should I do it?

**Comment by Tpt** [comment]
Thank you for using Oxigraph.

Sadly rdflib does not support RDF-star and rdflib API makes hard to swap rdflib parsers with Oxigraph parsers that do support RDF-star. However, you can hack a bit with code like:
```python
store = oxrdflib.OxigraphStore()
graph = rdflib.Graph(store=store) # Build the rdflib Graph object backed by the oxigraph store, you can also use ConjunctiveGraph or Dataset
store_inner.load(MY_FILE.ttl, "text/turtle", to_graph=pyoxigraph.BlankNode(graph.identifier)) # Loads the turtle file with Oxigraph parser supporting RDF-star, the `to_graph` parameter is here because rdflib expects the graph triples to be in the name graph. Beware, if you use a custom identifier, it might be a NamedNode and not a BlankNode
# The graph should contains the loaded RDF-star triples
```
Note that I have not tested it, syntax might be slightly wrong and accessing the data after loading might also fail. If you have the option I would suggest to use directly [pyoxigraph](https://pyoxigraph.readthedocs.io/en/stable/) instead.

**Reply by joylix** [reply]
I am not sure if pyoxigraph or Oxrdflib supports the described operation.Because the above two APIs do not have an open discussion area, so I am asking this question here. Thank you for your reply！
